addons/website_sale/models/res_partner.py

```diff
@@ -44,3 +44,21 @@ def _onchange_property_product_pricelist(self):
                     "Also, the cart might not be visible for the customer until you update the pricelist of that cart."
                 ),
             }}
+
+    def write(self, vals):
+        if {'city', 'zip', 'country_id', 'vat'} & vals.keys():
+            # Recompute fiscal position for open website orders
+            orders = self.env['sale.order'].sudo().search([
+                ('state', '=', 'draft'),
+                ('website_id', '!=', False),
+                '|', ('partner_id', 'in', self.ids), ('partner_shipping_id', 'in', self.ids),
+            ])
+            if orders:
+                fpos_by_order = {so: so.fiscal_position_id for so in orders}
```

Comment:
```suggestion
                fpos_by_order = {so.id: so.fiscal_position_id.id for so in orders}
```
nitpicking, but I prefer comparing and storing ids unless it's really useful to use records.

Reply:
Was thinking about using `orders_by_fpos = orders.grouped('fiscal_position_id')` starting from 17.0, or do you prefer keeping it like this?

Reply:
Fair enough. Not sure about 17.0, but 18.0 will require modifications regardless because of the `tax_id -> tax_ids` field rename on `sale.order.line`.